### PR TITLE
Add build matrix, and build for sbt 2.0

### PR DESCRIPTION
**Pull request by BillyAutrey**
* Adds a CI matrix
* Adds ability to build sbt 2.x and 1.x with the matrix
* Switches to temurin and the `actions/setup-java@v4` setup.
* Switches to checkout@v4

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,26 +7,49 @@ on:
...
         find $HOME/.cache/coursier/v1                -name "ivydata-*.properties" -delete || true
         find $HOME/.sbt
+    - name: Build and test (sbt 2.x)
+      if: ${{ matrix.sbt == 2 }}
+      shell: bash
+      run: |
+        gpg --import test-key.gpg
+        sbt -v ++3.x clean test scripted
+        rm -rf "$HOME/.ivy2/local" || true
+        find $HOME/Library/Caches/Coursier/v1        -name "ivydata-*.properties" -delete || true
```
Comment (BillyAutrey) [inline .github/workflows/ci.yml:52]: Could use a bit of insight into what this is doing. Feels a bit like they're trying to clean up specific ivy release data. Do we actually care here? Doesn't seem like anyone else is using this.

Comment (eed3si9n) [reply .github/workflows/ci.yml:52]: I don't know if any of these `find .. -delete` ever mattered.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,26 +7,49 @@ on:
...
     - name: Checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
     - name: Setup Scala
-      uses: olafurpg/setup-scala@v10
+      uses: actions/setup-java@v4
       with:
-        java-version: "adopt@1.8"
+        distribution: "${{ matrix.distribution }}"
+        java-version: "${{ matrix.java }}"
```
[the remark this answers is not in the document]
Comment (BillyAutrey) [reply .github/workflows/ci.yml:31]: Just to be clear, this means I can remove this?
```
    - name: Coursier cache
      uses: coursier/cache-action@v5
```

Should I also throw it in `publish.yml`?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (BillyAutrey) [reply .github/workflows/ci.yml]: Thanks, stripped it out.

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (BillyAutrey) [reply .github/workflows/ci.yml:31]: Done.

[the remark this answers is not in the document]
Comment (eed3si9n) [reply .github/workflows/ci.yml:31]: Yea. setup-java is so good it subsumes two actions.

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**